main.go

```diff
@@ -114,12 +126,64 @@ func handleBadge(w http.ResponseWriter, r *http.Request) {
 	buf.WriteTo(w)
 }
 
+// save the cookies for any upstream calls to the Ory apis
+func withCookies(ctx context.Context, v string) context.Context {
+	return context.WithValue(ctx, "req.cookies", v)
+}
+
+func getCookies(ctx context.Context) string {
+	return ctx.Value("req.cookies").(string)
+}
+
+// save the session to display it on the dashboard
+func withSession(ctx context.Context, v *ory.Session) context.Context {
+	return context.WithValue(ctx, "req.session", v)
+}
+
+func getSession(ctx context.Context) *ory.Session {
+	return ctx.Value("req.session").(*ory.Session)
+}
+
+func (app *App) sessionMiddleware(next http.HandlerFunc) http.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request) {
+		// set the cookies on the ory client
+		var cookies string
+
+		// this example passes all request.Cookies
+		// to `ToSession` function
+		//
+		// However, you can pass only the value of
+		// ory_session_projectid cookie to the endpoint
+		cookies = request.Header.Get("Cookie")
+
+		// check if we have a session
+		session, _, err := app.ory.FrontendApi.ToSession(request.Context()).Cookie(cookies).Execute()
+		if (err != nil && session == nil) || (err == nil && !*session.Active) {
```

Comment:
Double check the error handling here. Maybe log the error to see whats wrong

Reply:
I added logging to session and err and the session is always `<nil>`, so there is no error but FrontendApi.ToSession sets session as `<nil>`?

Do you think it could be related to the local Docker/tunnel setup?
Does it make sense for me to deploy this separately (as staging env) with a dummy slack connected?